builder/builder.go

```diff
@@ -201,83 +201,58 @@ func (b *Builder) SignManifestMoM() error {
 	return nil
 }
 
-// UpdateRepo will fetch the clr-bundles for our configured Clear Linux version
-func (b *Builder) UpdateRepo(ver string, allbundles bool) {
-	// Make the folder to store all clr-bundles version
-	if _, err := os.Stat("clr-bundles"); err != nil {
-		if err = os.Mkdir("clr-bundles", 0777); err != nil {
-			fmt.Fprintf(os.Stderr, "ERROR: Failed to create clr-bundles: %s\n", err.Error())
-		}
-	}
+const (
+	upstreamBundlesBaseDir   = ".mixer/upstream-bundles/"
+	upstreamBundlesVerDir    = "clr-bundles-%s"
+	upstreamBundlesBundleDir = "bundles"
+)
 
-	repo := "clr-bundles/clr-bundles-" + ver + ".tar.gz"
-	if _, err := os.Stat(repo); err == nil {
-		fmt.Println("Already downloaded " + repo)
-		return
-	}
+func getUpstreamBundlesDir(ver string) string {
+	return filepath.Join(upstreamBundlesBaseDir, fmt.Sprintf(upstreamBundlesVerDir, ver), upstreamBundlesBundleDir)
+}
 
-	URL := "https://github.com/clearlinux/clr-bundles/archive/" + ver + ".tar.gz"
-	err := helpers.Download(repo, URL)
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "ERROR: Failed to download clr-bundles, make sure the version is valid: %s\n", err)
-		os.Exit(1)
+func (b *Builder) getUpstreamBundles(ver string, prune bool) error {
+	// Make the folder to store upstream bundles if does not exist
+	if _, err := os.Stat(upstreamBundlesBaseDir); os.IsNotExist(err) {
+		if err = os.MkdirAll(upstreamBundlesBaseDir, 0777); err != nil {
```

Comment:
MkdirAll return nil if the path already exists, so you don't need the Stat call.

Reply:
Ah, that's a nice simplification.

Reply:
Fixed in forthcoming force push.